### PR TITLE
Add logs for message round trip traceability

### PR DESCRIPTION
**Pull request by dushaniw**
### Proposed changes in this pull request

- The debug logs added in this PR are required to identify the message hops and their latencies in each node during an API invocation in WSO2 APIM.
- [Documentation](https://docs.google.com/document/d/10-p1B9bORXHMG71XQfELxFQc0uHUOxlfD6TsvZBhuPs/edit?usp=sharing)

### When should this PR be merged

ASAP

### Follow up actions

- None

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 1]
```diff
@@ -686,7 +696,15 @@ private boolean isAnyPolicyContentAware(Connection conn, String apiPolicy, Strin
             ps.setInt(10, subscriptionTenantId);
             ps.setString(11, appPolicy);
             ps.setInt(12, appTenantId);
+            if(log.isDebugEnabled()){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Please format the code

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -686,7 +696,15 @@ private boolean isAnyPolicyContentAware(Connection conn, String apiPolicy, Strin
...
+                log.debug("Execution of isAnyPolicyContentAware sql query for apiId:"+ apiId + " started at "
+                        + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new java.util.Date()));
+            }
             resultSet = ps.executeQuery();
+            if(log.isDebugEnabled()){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Please format the code

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 3]
```diff
@@ -5554,7 +5572,15 @@ public void updateURLTemplates(API api) throws APIManagementException {
             prepStmt.setString(1, apiContext);
             prepStmt.setString(2, version);
 
+            if(log.isDebugEnabled()){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Please format the code

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 4]
```diff
@@ -5554,7 +5572,15 @@ public void updateURLTemplates(API api) throws APIManagementException {
...
+                log.debug("Execution of getAllURITemplatesOldThrottle sql query for:"+ apiContext + " started at "
+                        + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new java.util.Date()));
+            }
             rs = prepStmt.executeQuery();
+            if(log.isDebugEnabled()){
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Please format the code

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java [hunk 5]
```diff
@@ -181,7 +182,17 @@ private void trackPageView(MessageContext msgCtx) throws Exception {
...
+                    + " started" + " at "
+                    + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new Date()));
+        }
         GoogleAnalyticsDataPublisher.publishGET(payload, userAgent, false);
+        if (log.isDebugEnabled()) {
+            log.debug("Publishing https GET from gateway to Google analytics" + " with ID: " + msgCtx.getMessageID()
+                    + " ended" + " at "
+                    + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new Date()));
+        }
 	}
```
Comment (dinusha92) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java]: Seems to have a formatting issue here. Could you please check it?

Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java]: @dinusha92  Fixed the formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtFaultHandler.java [hunk 6]
```diff
@@ -90,7 +93,17 @@ public boolean mediate(MessageContext messageContext) {
...
+            if (log.isDebugEnabled()) {
+                log.debug("Publishing fault event from gateway to analytics for: " + messageContext.getProperty(
+                        APIMgtGatewayConstants.CONTEXT) + " with ID: " + messageContext.getMessageID() + " started"
+                        + " at " + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new Date()));
+            }
             publisher.publishEvent(faultPublisherDTO);
+            if (log.isDebugEnabled()) {
+                log.debug("Publishing fault event from gateway to analytics for: " + messageContext.getProperty(
+                        APIMgtGatewayConstants.CONTEXT) + " with ID: " + messageContext.getMessageID() + " ended"
+                        + " at " + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new Date()));
```
[the remark this answers is not in the document]
Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtFaultHandler.java]: Thank you for pointing it out. Will work on defining constants for that.

**Pull request merged**